tests/nestedlib/parsers/PyJinjaParserTest.py

```diff
@@ -0,0 +1,286 @@
+        # If the pure line has spaces before the element and after the element
+        # line = '    {{ var }}    \n'
+
+        line = '    {{ var }}    \n'
+        uut_nl_section = self.parser.parse_line(line=line,
+                                                nl_sections=[],
+                                                line_number=1,
+                                                file=self.test_filename1)
+
+        expected_nl_section_1_str = ('.*test-jinja-py.py.jj2.txt' +
+                                     ': 1: jinja: L1 C1: L1 C17: L1 C1: L1 C17')
+        self.assertRegex(str(uut_nl_section[0]), expected_nl_section_1_str)
+
+    def test_coala_setup_template(self):
+
+        # Test the setup template of coala present in the moban repository
+        # The expected output of the parser is saved in another file.
+
+        test_filename = 'test-setup-py.py.jj2.txt'
+        test_file_path = os.path.join(self.file_test_dir, test_filename)
+        abs_test_file_path = abspath(test_file_path)
+
+        test_output_filename = 'coala-setup-parser-output.txt'
+        test_file_ouput_path = os.path.join(self.file_test_output,
+                                            test_output_filename)
+        abs_test_file_output_path = abspath(test_file_ouput_path)
+        test_ouput_file = File(abs_test_file_output_path)
+
+        uut_sections = self.parser.parse(abs_test_file_path)
+        nl_section_string = ''
+        expected_string = ''
+
+        for nl_section in uut_sections:
+            nl_section_string += (str(nl_section)+'\n')
+
+        for line in test_ouput_file.lines:
+            expected_string += (abs_test_file_path + line)
+
+        self.assertEqual(nl_section_string, expected_string)
+
+    def test_coala_setup_template_output(self):
+
+        # Generates text using the values generated by the parser
+        # Check if the generated text is equal to the original file
+
+        test_filename = 'test-setup-py.py.jj2.txt'
+        test_file_path = os.path.join(self.file_test_dir, test_filename)
+        abs_test_file_path = abspath(test_file_path)
+
+        original_file = File(abs_test_file_path)
+        original_file_content = original_file.string
+
+        uut_sections = self.parser.parse(abs_test_file_path)
+        generated_file_contents = ''
+
+        for nl_section in uut_sections:
+
+            start_line = nl_section.start.line
+            start_column = nl_section.start.column
+            end_line = nl_section.end.line
+            end_column = nl_section.end.column
+
+            for line_nr in range(start_line, end_line+1):
+                if (line_nr == end_line and line_nr == start_line):
+                    generated_file_contents += (
+                        original_file[line_nr-1][start_column-1:end_column])
+                    if end_column == len(original_file[line_nr-1]) - 1:
+                        generated_file_contents += "\n"
```

Comment:
use single-quotes to comply with QuotesBear. Otherwise tests won't pass

Reply:
Thanks! Tests are passing now :)